### PR TITLE
[fix](auth)support check priv when tvf use resource

### PR DESCRIPTION
**Pull request by zddr**
current, follow sql not check priv, we need check `usage priv` of `test_resource`
```
SELECT * FROM S3 (
    "uri" = "xxx",
    "format" = "orc",
    "resource" = "test_resource",
    "use_path_style" = "true"
);
````

### CODE REVIEW

**Review by 924060929 — DISMISSED** [review]

**Comment by zddr** [comment]
run buildall

File: regression-test/suites/external_table_p0/tvf/test_s3_tvf_with_resource.groovy [hunk 1]
```diff
@@ -176,4 +176,24 @@ suite("test_s3_tvf_with_resource", "p0") {
     } finally {
     }
 
+    // test auth
+    String user = 'test_s3_tvf_with_resource_user'
+    String pwd = 'C123_567p'
+    try_sql("DROP USER ${user}")
+    sql """CREATE USER '${user}' IDENTIFIED BY '${pwd}'"""
```
Comment (924060929) [inline regression-test/suites/external_table_p0/tvf/test_s3_tvf_with_resource.groovy:183]: add cloud mode grant statement
```groovy

    //cloud-mode
    if (isCloudMode()) {
        def clusters = sql " SHOW CLUSTERS; "
        assertTrue(!clusters.isEmpty())
        def validCluster = clusters[0][0]
        sql """GRANT USAGE_PRIV ON CLUSTER ${validCluster} TO ${user}""";
    } 
```

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by BePPPower — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**